### PR TITLE
Add support for dynamic keyframes

### PR DESCRIPTION
**Pull request by HenriBeck**
__What would you like to add/fix?__
This fixes having a dynamic animation name.

I think we can have a better system to process the update style changes and not require this duplicate code.

Still missing some tests to make sure we don't break anything.

### CODE REVIEW

**Comment by HenriBeck** [comment]
@kof please review.

File: packages/jss/src/plugins/keyframesRule.js [hunk 1]
```diff
@@ -82,36 +81,66 @@ const keyRegExp = /@keyframes\s+/
...
+  if (refKeyframe !== value) {
+    style[prop] = refKeyframe
   }
 }
 
-export default {
-  onCreateRule(key: string, frames: JssStyle, options: RuleOptions): KeyframesRule | null {
-    return keyRegExp.test(key) ? new KeyframesRule(key, frames, options) : null
+const plugin: Plugin = {
+  onCreateRule(key, frames, options) {
```
Comment (kof) [inline packages/jss/src/plugins/keyframesRule.js]: why removing the types?

Comment (kof) [reply packages/jss/src/plugins/keyframesRule.js]: ah gotcha

File: packages/jss/src/plugins/keyframesRule.js [hunk 2]
```diff
@@ -82,36 +81,66 @@ const keyRegExp = /@keyframes\s+/
...
 
     if ('animation-name' in style) replaceRef(style, 'animation-name', sheet.keyframes)
     if ('animation' in style) replaceRef(style, 'animation', sheet.keyframes)
     return style
+  },
+
+  onChangeValue(val, prop, rule) {
+    const {sheet} = rule.options
+
+    if (rule.type !== 'style' || !sheet) {
```
Comment (kof) [inline packages/jss/src/plugins/keyframesRule.js]: afaik only style rule value can be changed, so the first check can be removed

**Pull request merged**